### PR TITLE
fix(iot-dev): Upgrade mqtt Paho dependency version

### LINKED ISSUE #596: [Java 1.7] MQTT runtime exception due to paho 1.2.2
Details on the paho issue [here](https://github.com/eclipse/paho.mqtt.java/issues/633)

The latest release of the java device client/deps packages takes a dependency on paho 1.2.2 which encounters a runtime exception while opening a connection using mqtt.

This exception seems to be limited to mqtt users below Android 7, as well as Java SDK version 1.7 and below. JDK 1.8+ works as expected.

### PR DESCRIPTION
**Pull request by timtay-microsoft**
Fixes #596

Here are the release notes from this new Paho version: https://github.com/eclipse/paho.mqtt.java/releases/tag/v1.2.3

### CODE REVIEW

File: device/iot-device-client/pom.xml [hunk 1]
```diff
@@ -65,7 +65,7 @@
         <dependency>
             <groupId>org.eclipse.paho</groupId>
             <artifactId>org.eclipse.paho.client.mqttv3</artifactId>
-            <version>1.2.2</version>
```
Comment (abhipsaMisra) [inline device/iot-device-client/pom.xml:68]: Question - doesn't device-client take a dependency on deps? Why do we have this added in both?

Comment (timtay-microsoft) [reply device/iot-device-client/pom.xml:68]: Good question! We don't need it declared in device client's pom, so I'll get rid of it.

**Comment by timtay-microsoft** [comment]
Trying out this library locally, and I'm seeing some unexpected behavior when running Java 7 or lower still. Don't merge this in yet

**Comment by timtay-microsoft** [comment]
Closing this PR as the upgraded paho version didn't fix any bugs that we had

**Pull request closed without merging**